sdk/core/azure-json/src/main/java/com/azure/json/JsonDataStructure.java

```diff
@@ -0,0 +1,16 @@
+package com.azure.json;
+
+import java.io.IOException;
+
+public abstract class JsonDataStructure extends JsonElement {
+    @Override
+    public String toString() {
+        return null;
+    }
+
+    public abstract void build(JsonReader reader) throws IOException;
```

Comment:
This type may not be needed as we want `JsonReader` to be in the constructor of the JsonArray JsonObject. Also, toJson is not needed as JsonElement already has `serialize` that can then produce the json.